### PR TITLE
P#54339 Problem with similar estates and the displaying of API error message (Prio)

### PR DESCRIPTION
**Pull request by vu-vuong-esg**
Sending "georangesearch" parameter to the API caused the problem so I removed that parameter.
I also saw that in the similar estate's dashboard the "georangesearch" parameter (Disabled in onOffice)

### CODE REVIEW

**Comment by jayay** [comment]
it was added in df02975bb9dc42ff66a85ba9f298844785ce0a65 because the radius search was broken for similar estates. I think it should be repaired instead of removed in order to get the feature to work.

**Comment by michaelarnold-dev** [comment]
@hungnc89 : The testers observed the following: Could you please check? "With version 2.19.7-3-gc8d3054a installed, I get the Uncaught TypeError from the T.I. when entering some estate lists in the backend.

Example: [oo_estate view="Immobilienliste"] in https://wpptest.onofficeweb.com/"

**Comment by tung-le-esg** [comment]
> @hungnc89 : @tung-le-esg The testers observed the following: Could you please check? "With version 2.19.7-3-gc8d3054a installed, I get the Uncaught TypeError from the T.I. when entering some estate lists in the backend.
> 
> Example: [oo_estate view="Immobilienliste"] in https://wpptest.onofficeweb.com/"
> 
> Fatal error: Uncaught TypeError: Argument 1 passed to onOffice\WPlugin\DataView\DataListView::setSortByUserDefinedDirection() must be of the type int, string given, called in /www/htdocs/w01b98d7/227.wp.myonoffice.com/htdocs/wp-content/plugins/onoffice/plugin/DataView/DataListViewFactory.php on line 76 and defined in /www/htdocs/w01b98d7/227.wp.myonoffice.com/htdocs/wp-content/plugins/onoffice/plugin/DataView/DataListView.php:314 Stack trace: #0 /www/htdocs/w01b98d7/227.wp.myonoffice.com/htdocs/wp-content/plugins/onoffice/plugin/DataView/DataListViewFactory.php(76): onOffice\WPlugin\DataView\DataListView->setSortByUserDefinedDirection('') #1 /www/htdocs/w01b98d7/227.wp.myonoffice.com/htdocs/wp-content/plugins/onoffice/plugin/Gui/AdminPageEstateListSettings.php(249): onOffice\WPlugin\DataView\DataListViewFactory->createListViewByRow(Array) #2 /www/htdocs/w01b98d7/227.wp.myonoffice.com/htdocs/wp-content/plugins/onoffice/plugin/Gui/AdminPageEstateListSettingsBase.php(54): onOffice\WPlugin\Gui\AdminPageEstateListSettings->val in /www/htdocs/w01b98d7/227.wp.myonoffice.com/htdocs/wp-content/plugins/onoffice/plugin/DataView/DataListView.php on line 314

Okay. I will check the error.

File: plugin/DataView/DataListViewFactory.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ public function createListViewByRow(array $row): DataViewFilterableFields
 		$pListView->setAvailableOptions($row['availableOptions']);
 		$pListView->setSortBySetting((int)$row['sortBySetting']);
 		$pListView->setSortByUserDefinedDefault($row['sortByUserDefinedDefault']);
-		$pListView->setSortByUserDefinedDirection($row['sortByUserDefinedDirection']);
+		$pListView->setSortByUserDefinedDirection((int) $row['sortByUserDefinedDirection']);
```
Comment (jayay) [inline plugin/DataView/DataListViewFactory.php:76]: Shouldn't this be `(bool)`?

Comment (tung-le-esg) [reply plugin/DataView/DataListViewFactory.php:76]: This field is enum('0','1') data type and return string when we select it from the database.

Comment (jayay) [reply plugin/DataView/DataListViewFactory.php:76]: 👍🏻

**Review by jayay — APPROVED** [review]

**Pull request merged**